mesheryctl/internal/cli/root/mesh/mesh.go

```diff
@@ -1,41 +1,310 @@
+	}
+
+	// get a list of meshes from available adapters
+	meshNames := []string{}
+	for _, adapter := range prefs.MeshAdapters {
+		meshNames = append(meshNames, adapter.Name)
+	}
+
+	// return an error if no adapters were found
+	if len(meshNames) == 0 {
+		return "", ErrNoAdapters
+	}
+
+	// allow the user to select a mesh from the list of available ones
+	prompt := promptui.Select{
+		Label: "Select a Service Mesh from the list",
+		Items: meshNames,
+	}
+	i, _, err := prompt.Run()
+	if err != nil {
+		return "", ErrPrompt(err)
+	}
+
+	// return the selected mesh name
+	return meshNames[i], nil
+}
+
+func sendOperationRequest(mctlCfg *config.MesheryCtlConfig, query string, delete bool, spec string) (string, error) {
+	path := mctlCfg.GetBaseMesheryURL() + "/api/system/adapter/operation"
+	method := "POST"
+	data := url.Values{}
+	data.Set("adapter", adapterURL)
+	data.Set("query", query)
+	data.Set("customBody", "")
+	data.Set("namespace", namespace)
+	if delete {
+		data.Set("deleteOp", "on")
+	} else {
+		data.Set("deleteOp", "")
+	}
+
+	// Choose which specification to use for conformance test
+	// Used in validate command
+	switch spec {
+	case "smi":
+		{
+			data.Set("query", "smi_conformance")
+			break
+		}
+	case "istio-vet":
+		{
+			if adapterURL == "meshery-istio:10000" {
+				data.Set("query", "istio-vet")
+				break
+			}
+			return "", errors.New("only Istio supports istio-vet operation")
+		}
+	case "null":
+		{
+			// If validate command isn't used
+			// This must be used only for deploy or remove operations
+			break
+		}
+	default:
+		{
+			return "", errors.New("specified specification not found or not yet supported")
+		}
+	}
+
+	payload := strings.NewReader(data.Encode())
+
+	client := &http.Client{}
+	req, err := utils.NewRequest(method, path, payload)
+	if err != nil {
+		return "", ErrCreatingValidateRequest(err)
+	}
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded;charset=UTF-8")
+
+	res, err := client.Do(req)
+	if err != nil {
+		return "", ErrCreatingDeployRequest(err)
+	}
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return "", err
+	}
+	return string(body), nil
+}
+
+func waitForDeployResponse(mctlCfg *config.MesheryCtlConfig, query string) (string, error) {
+	path := mctlCfg.GetBaseMesheryURL() + "/api/events?client=cli_deploy"
+	method := "GET"
+	client := &http.Client{}
+	req, err := utils.NewRequest(method, path, nil)
+	if err != nil {
+		return "", ErrCreatingDeployRequest(err)
+	}
+
+	res, err := client.Do(req)
+	if err != nil {
+		return "", ErrCreatingDeployResponseRequest(err)
+	}
+	defer res.Body.Close()
+
+	event, err := utils.ConvertRespToSSE(res)
+	if err != nil {
+		return "", ErrCreatingDeployResponseStream(err)
+	}
+
+	timer := time.NewTimer(time.Duration(1200) * time.Second)
+	eventChan := make(chan string)
+
+	//Run a goroutine to wait for the response
+	go func() {
+		for i := range event {
+			if strings.Contains(i.Data.Details, query) {
```

Comment:
can you point me to valid API response for any deployment(install/remove) request?

Reply:
The message after request won't be printed... still...

https://github.com/alphaX86/meshery/blob/9e69993e7ac0afcc3d604441182f8cce3e544bdc/mesheryctl/internal/cli/root/mesh/mesh.go#L152-L154

I refactored the above function to support install/remove/validate operations

Reply:
so for the deployment case, you might not have a query field in the payload, so why are you checking for it in the waitForDeployResponse() ?

Reply:
If --watch is applied

Reply:
got it